src/lib/core/adaptive/screen-type.ts

```diff
@@ -0,0 +1,134 @@
+import {Injectable, NgZone} from '@angular/core';
+import {Observable} from 'rxjs/Observable';
+import {BehaviorSubject} from 'rxjs/BehaviorSubject';
+
+
+export const SCREEN_TYPES = {
+  HANDSET: '(max-width: 600px) and (orientation: portrait), (max-width: 960px) and ' +
+           '(orientation: landscape)',
+  TABLET: '(min-width: 600px) and (max-width: 840px) and (orientation: portrait), ' +
+          '(min-width: 960px) and (max-width: 1280px) and (orientation: landscape)',
+  WEB: '(min-width: 840px) and (orientation: portrait), (min-width: 1280px) and ' +
+       '(orientation: landscape)',
+
+  HANDSET_PORTRAIT: '(max-width: 600px) and (orientation: portrait)',
+  TABLET_PORTRAIT: '(min-width: 600px) and (max-width: 840px) and (orientation: portrait)',
+  WEB_PORTRAIT: '(min-width: 840px) and (orientation: portrait)',
+
+  HANDSET_LANDSCAPE: '(max-width: 960px) and (orientation: landscape)',
+  TABLET_LANDSCAPE: '(min-width: 960px) and (max-width: 1280px) and (orientation: landscape)',
+  WEB_LANDSCAPE: '(min-width: 1280px) and (orientation: landscape)',
+};
+
+export interface Breakpoint {
+  observe: Observable<boolean>;
+  destroy: () => void;
+}
+
+interface InternalBreakpoint {
+  query: string;
+  subject: BehaviorSubject<boolean>;
+  callback: (mql: MediaQueryList) => void;
+  usageCount: number;
+  watcher: MediaQueryList;
+}
```

Comment:
Naming could probably be better here, but essentially InternalBreakpoint is an object containing all of the information needed to create a Breakpoint to be handed out.